design/src/client/detailed_error_explanations.md

```diff
@@ -0,0 +1,31 @@
+# Detailed Error Explanations
+
+This page collects detailed explanations for some errors. If you encounter an error
+and are interested in learning more about what it means and why it occurs, check here.
+
+**If you can't find the explanation on this page, please file an issue asking for it to be added.**
+
+## "Connection encountered an issue and should not be re-used. Marking it for closure"
+
+The SDK clients each maintain their own connection pool. By the convention of some services,
```

Comment:
This isn't strictly true, in fact most of the time connection pools are shared across clients. Depends on how the client is constructed though.